ls1intum/Artemis
# `General`: Forward changes to internal users to Bitbucket

### Checklist
#### General


- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.(https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
#### Server

### Motivation and Context


If a password gets changed, changes only get delegated to GitLab/Jenkins but not Bitbucket so the password change only affects Artemis.

### Description

I implemented the BitbucketUser ManagementService and moved the Bitbucket user creation from participation initialisation to Artemis user creation as also suggested by a TODO.
I also changed it so all internal users can access the password reset.

### Steps for Testing

Prerequisites:
- 1 Testserver with Atlassian connected
- 1 Student with a local user Account
- 1 Programming Exercise 
**You need a local user for this. If you already have one, please use it, if you need one, please let me know.**

1. Log in to Artemis
2. Start your participation to the exercise
3. Download the repository with the students credentials and push changes
4. Change your password using the change password function in the top-right menu
5. Make sure you can't do any changes to the repository with your old password but you can with your new password.
6. Log out and reset the password using the password reset function
7. Make sure you can't do any changes to the repository with your old password but you can with your new password.
8. Change the password back to the default

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage





BitbucketUserManagementService: 69.44% | 94.44%
BitbucketService: 44.19% | 66.39%
BitbucketAuthorizationInterceptor.java: 73.33% | 92.86%
GitLabService.java: 45.65% | 83.60%
JiraAuthenticationProvider.java: 37.50% | 66.67%
UserCreationService.java: 69.64% | 94.04%
AccountResource.java: 77.08% | 95.65%
CourseResource.java: 73.08% | 95.42%
